### DISCUSSION TITLE
[Issue] SecurityTrails showing same subdomains for "x.in.net" and "in.net"

### DISCUSSION THREAD

**Discussion by xscorp** [projectdiscovery/subfinder#856] [Q&A] [answered]
**Issue**
The SecurityTrails API appears to show same subdomains for `bhim.in.net` and `in.net`. 

This is not a problem with subfinder, but with the SecurityTrails API. Probable cause of this issue could be bad parsing of domains from subdomains or using a non-reliable list of public suffixes.

I request the maintainers to verify this issue, and temporarily disable the `securitytrails` source for subdomains ending with ".in.net" by default unless the issue is fixed by SecurityTrails team.

**Evidence**

<img width="762" alt="image" src="https://user-images.githubusercontent.com/37046662/213125413-551d53a4-e436-48f2-a9ce-3f165d8a261d.png">

As can be seen in the above screenshot, `securitytrails` source is returning same subdomains for `in.net` TLD and the domain `bhim.in.net`.

**Comment by mr-rizwan-syed** [comment]
subfinder shows double subdomain like below, might be due to securitytrails.

sftp.domain.com.domain.com
content.domain.com.domain.com
www.domain.com.domain.com

I commented securtytrails from provider-config file and it resolved

**Comment by ehsandeep** [comment, marked as answer]
@xscorp This looks like edge case in SecurityTrails API; instead of disabling it for all, we suggest to disable this source in your config file.

**Comment by ehsandeep** [comment]
@mr-rizwan-syed, thanks for pointing this, it should be fixed with https://github.com/projectdiscovery/subfinder/pull/849

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
